File: Cpp/929_Unique_Email_Addresses.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        unordered_set<string> s;
        
        for(auto& email : emails){
            string local_name = "";
            int pos = 0;
            for(; pos<email.size(); ++pos){
                if(email[pos] == '+'){
                    while(email[pos] != '@') {
                        ++pos;
                    }
                    break;
                } else if(email[pos] == '@') {
                    break;
                } else if(email[pos] != '.') {
                    local_name += email[pos];
                }
            }
            
            s.insert(local_name + email.substr(pos));
        }
        
        return s.size();
    }
};

File: Kotlin/929_Unique_Email_Addresses.kt
class Solution {
    fun numUniqueEmails(emails: Array<String>): Int {	
		val s = mutableSetOf<String>()
        
		for(email in emails) {
			val splitEmail = email.split('@')
			val localName = splitEmail[0].replace(".", "").split("+")[0]
			val domainName = splitEmail[1]
			
			s.add("$localName@$domainName")
		}
		
		return s.size
    }
}